api/src/main/java/org/openmrs/module/locationbasedaccess/PatientSearchAdviser.java

```diff
@@ -0,0 +1,86 @@
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.module.locationbasedaccess;
+
+
+import org.aopalliance.aop.Advice;
+import org.aopalliance.intercept.MethodInvocation;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.*;
+import org.openmrs.api.context.Context;
+import org.springframework.aop.Advisor;
+import org.springframework.aop.support.StaticMethodMatcherPointcutAdvisor;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class PatientSearchAdviser extends StaticMethodMatcherPointcutAdvisor implements Advisor {
+
+    private static final Log log = LogFactory.getLog(PatientSearchAdviser.class);
+    public static final  String LOCATION_ATTRIBUTE_GLOBAL_PROPERTY_NAME = "locationbasedaccess.locationAttributeUuid";
+
+    @Override
+    public boolean matches(Method method, Class targetClass) {
+        if (method.getName().equals("getPatients")) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public Advice getAdvice() {
+        return new PatientSearchAdvise();
+    }
+
+    private class PatientSearchAdvise implements MethodInterceptor {
+
+        public Object invoke(MethodInvocation invocation) throws Throwable {
+            Integer sessionLocationId = Context.getUserContext().getLocationId();
+            String locationAttributeUuid = Context.getAdministrationService().getGlobalProperty(LOCATION_ATTRIBUTE_GLOBAL_PROPERTY_NAME);
+            List<Patient> patientList = (List<Patient>) invocation.proceed();
+            if (StringUtils.isNotBlank(locationAttributeUuid)) {
+                final PersonAttributeType personAttributeType = Context.getPersonService().getPersonAttributeTypeByUuid(locationAttributeUuid);
+                if (sessionLocationId != null) {
+                    String sessionLocationUuid = Context.getLocationService().getLocation(sessionLocationId).getUuid();
+
+                    for (Iterator<Patient> iterator = patientList.iterator(); iterator.hasNext(); ) {
+                        Patient patient = iterator.next();
+                        if (!isPatientContainsLocation(sessionLocationUuid, patient.getAttribute(personAttributeType))) {
+                            iterator.remove();
+                        }
+                    }
+                } else {
+                    // If the sessionLocationId is null, then return a empty list
+                    log.debug("Search Patient : Null Session Location in the UserContext");
+                    return new ArrayList<Patient>();
+                }
+            }
+            return patientList;
+        }
+
+        public Boolean isPatientContainsLocation(String locationUuid, PersonAttribute personAttribute ) {
```

Comment:
Looking at the method name, is there anything in this method that has to do with patient?

Reply:
That's why I thought to replace by `isPatientAttributeContainsLocation()`. Wha do you think about this name?

Reply:
Can you just do a 10 minutes google search about good method names?